lib/Push.php

```diff
@@ -112,12 +128,44 @@ public function isDeferring(): bool {
 	}
 
 	public function deferPayloads(): void {
+		$this->deferPreparing = true;
 		$this->deferPayloads = true;
 	}
 
 	public function flushPayloads(): void {
-		$this->deferPayloads = false;
+		$this->deferPreparing = false;
+
+		if (!empty($this->loadDevicesForUsers)) {
+			$missingDevicesFor = array_diff($this->loadDevicesForUsers, array_keys($this->userDevices));
+			$this->loadDevicesForUsers($missingDevicesFor);
```

Comment:
I would suggest to define `getDevicesForUsers()` instead of `loadDevicesForUsers()`. It will be more similar to the existing `getDevicesForUser()` and it will make this block more similar to the block below for user statuses.